### PR TITLE
CONTRACTS: build a function to read user's position overview.

### LINKED ISSUE #45: CONTRACTS: build a function to read user's position overview.
after a proposal has been accepted in issue #43 and the loan has been sent, these open positions should be displayed on the users portfolio page in the position overview tab

Screenshot 2024-11-21 at 07 12 23

for borrowers, it should show the details of what was borrowed like:

token borrowed,
expected repayment time,
interest rate,
amount borrowed,
and a repay button.

build this function, write tests for it and integrate it into the frontend on the porfolio page.

### PR DESCRIPTION
**Pull request by Akshola00**
Fixes #45

## **[Feature] Display Borrower's Open Positions in Portfolio Page**

This PR implements the functionality to display a borrower’s open positions in the **Position Overview Tab** of the portfolio page, addressing **Issue #43**. Borrowing details include the token borrowed, expected repayment time, interest rate, amount borrowed, and a **Repay Button** to facilitate loan repayment. Backend updates include an API endpoint to fetch open positions and handle repayment logic, while the frontend integrates this data into the UI and dynamically updates the display after repayment.  

To test, verify the API endpoint for open positions and repayment functionality through backend tests and ensure data renders correctly on the portfolio page. Manual testing should confirm the display of borrowing positions.

### CODE REVIEW

**Review by Dprof-in-tech — CHANGES_REQUESTED** [review]
doesnt build

File: contracts/src/peer_protocol.cairo [hunk 1]
```diff
@@ -24,12 +24,22 @@ struct Transaction {
     tx_hash: felt252,
 }
 
+#[derive(Drop, Serde, Copy, starknet::Store)]
```
[the remark this answers is not in the document]
Comment (Akshola00) [reply contracts/src/peer_protocol.cairo:27]: resolving that now

**Comment by Akshola00** [comment]
it builds and merge conflicts have been resolved

**Pull request merged**